### ISSUE TITLE
Autoprimary/Autosecondary Pair

### ISSUE THREAD

**Issue by TheDarkula** [PowerDNS/pdns#13085]
- Program: Authoritative
 - Issue type: Bug report

### Short description
I am trying to configure an autoprimary/autosecondary pair, and hitting resolving issues.

### Environment

 - Operating system: Running pdns on Alpine Linux in docker/kubernetes/k3s
 - Software version: 4.7.4-r0
 - Software source: Alpine official packages

### Steps to reproduce

1. I created a master/primary zone on ns1.example.com.
2. In the autosecondary, I ran:
	pdnsutil add-autoprimary ns1public.ip.address.here ns2.example.com accountname

### Expected behaviour
The autosecondary to create/update the domain from the autoprimary.

### Actual behaviour
Relevant logs:
```
ns1:
	Jul 31 04:38:04 Queued notification of domain 'example.com' to ns1public.ip.address.here:53
	Jul 31 04:38:04 Queued notification of domain 'example.com' to ns2public.ip.address.here:53
	Jul 31 04:38:04 Received NOTIFY for example.com from 10.42.0.1 but slave support is disabled in the configuration
	Jul 31 04:38:05 Received unsuccessful notification report for 'example.com' from ns1public.ip.address.here:53, error: Query Refused
	Jul 31 04:38:05 Removed from notification list: 'example.com' to ns1public.ip.address.here:53 Query Refused
	Jul 31 04:38:05 Removed from notification list: 'example.com' to ns2public.ip.address.here:53 (was acknowledged)

ns2:
	Jul 31 04:38:04 Received NOTIFY for example.com from 10.42.0.1 for which we are not authoritative, trying supermaster
	Jul 31 04:38:08 Error resolving SOA or NS for example.com at: 10.42.0.1: Timeout waiting for answer from 10.42.0.1
```

### Other information
From what I can tell, the 10.42.0.1 address is from k3s/flannel, but I have not been able to figure out why that is showing up in the logs for a notify.

Another bit of detail is that I am using adguard home as my recursive resolver.

**Comment by omoerbeek** [comment]
We need unedited config files and logs.
Please read https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open.

**Comment by TheDarkula** [comment]
Ah, no worries :)

ns1 logs:
```
Aug 06 04:22:46 Queued notification of domain 'kincke.com' to 63.230.56.225:53
Aug 06 04:22:46 Queued notification of domain 'kincke.com' to 63.230.56.226:53
Aug 06 04:22:46 Received NOTIFY for kincke.com from 10.42.0.1 but slave support is disabled in the configuration
Aug 06 04:22:47 Received unsuccessful notification report for 'kincke.com' from 63.230.56.225:53, error: Query Refused
Aug 06 04:22:47 Removed from notification list: 'kincke.com' to 63.230.56.225:53 Query Refused
Aug 06 04:22:47 Removed from notification list: 'kincke.com' to 63.230.56.226:53 (was acknowledged)
```

ns2 logs:
```
Aug 06 04:21:46 Received NOTIFY for kincke.com from 10.42.0.1 for which we are not authoritative, trying supermaster
Aug 06 04:21:50 Error resolving SOA or NS for kincke.com at: 10.42.0.1: Timeout waiting for answer from 10.42.0.1
```

Also, here is a `drill` from inside the ns2 container:
```
$ drill SOA kincke.com
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 18604
;; flags: qr rd ra ; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; kincke.com.	IN	SOA

;; ANSWER SECTION:
kincke.com.	30	IN	SOA	ns1.kincke.com. hostmaster.kincke.com. 2023080607 10800 3600 604800 60

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 43 msec
;; SERVER: 10.43.0.10
;; WHEN: Sun Aug  6 04:27:27 2023
;; MSG SIZE  rcvd: 109
```

**Comment by Habbie** [comment]
> ;; SERVER: 10.43.0.10

that's not `10.42.0.1`, can you try `drill SOA kincke.com @10.42.0.1`?

(with that, we are closing this ticket, as it is clearly a support issue and not a bug report)

**Comment by TheDarkula** [comment]
@Habbie I finally sorted out the odd networking issue!
It was a NAT loopback internal vs. external problem.

Now though, I do feel like I have run into a bug, or something wholly undocumented.

I explained the whole situation in a discussion [here](https://github.com/PowerDNS/pdns/discussions/14071), but basically the autosecondary is querying itself for the SOA record instead of the autoprimary.

Is there some simple configuration option I am missing here?
